azure/services/scalesets/scalesets.go

```diff
@@ -271,10 +270,23 @@ func (s *Service) patchVMSSIfNeeded(ctx context.Context, infraVMSS *azure.VMSS)
 	}
 
 	hasModelChanges := hasModelModifyingDifferences(infraVMSS, vmss)
-	if maxSurge > 0 && (hasModelChanges || !infraVMSS.HasEnoughLatestModelOrNotMixedModel()) {
+	var isFlex bool
+	for _, instance := range infraVMSS.Instances {
+		if instance.InstanceID == "" {
+			isFlex = true
+			break
+		}
+	}
+	var hasEnoughLatestModelOrNotMixedModel bool
```

Comment:
Not sure what `hasEnoughLatestModelOrNotMixedModel ` means at first read. Consider adding a code comment to explain?